config/app_protocol.go

```diff
@@ -0,0 +1,24 @@
+package config
+
+type appProtocol string
+
+func (ap appProtocol) ApplyHTTP(cfg *httpOptions) {
+	cfg.commonOpts.ForwardsProto = string(ap)
+}
+
+func (ap appProtocol) ApplyLabeled(cfg *labeledOptions) {
+	cfg.commonOpts.ForwardsProto = string(ap)
+}
+
+// WithAppProtocol declares the protocol that the upstream service speaks.
+// This may be used by the ngrok edge to make decisions regarding protocol
+// upgrades or downgrades.
+//
+// Currently, `http2` is the only valid string, and will cause connections
```

Comment:
This mentions `http2` as the only legal value, but the proto msg has `ForwardsProto string    // the L7 protocol the upstream service is expecting (one of '', 'h1', 'h2c')`. I don't think there's any translation or validation going on in here, so just wanted to double-check if either of these comments might be incorrect.

Reply:
heh, what niki said.